### DISCUSSION TITLE
I improved my 8-bit computer.

### DISCUSSION THREAD

**Discussion by megarcher** [SebLague/Digital-Logic-Sim#155] [Show and tell]
In my last post (https://itch.io/t/2624472/i-made-an-8-bit-computer-in-digital-logic-sim-v2), I showed a simple self-made CPU. (almost completely my own architecture!) since then, I've learned a lot about CPU's. Yesterday I finally finished building my new 8-bit computer, so today I thought, why not make a post about it.
Let's start with some photo's
![image](https://user-images.githubusercontent.com/62549925/219943194-7d0dfab5-2e72-4bd0-b3b5-bf895449dad8.png)
This is the finished product. The blue wires are clock signals, the yellows are instruction signals. There's an 8-bit BUS in the middle of the CPU, which connects all chips together. It's an address and data bus, depending on the instruction.
![image](https://user-images.githubusercontent.com/62549925/219943490-4a75ed65-830a-4b77-80aa-72cfb22f5823.png)
These are all the instructions that are implemented.
The first test I did was creating a program that shows all the multipliers of 3. This is the ROM I used:
address, instruction, data(optional)
0000 | ADD | 0100
0001 | OUT | 0000
0010 | JMC | 0000
0011 | JMP | 0000
0100 | 0000 | 0101
It worked first try, thankfully.
Then I tried the following ROM:
0000 | MVA | 0001
0001 | STA | 1111
0010 | STB | 1110
0011 | ADD | 1110
0100 | LDB | 1111
0101 | OUT | 0000
0110 | JMP | 0001
Which did not work first try.. But after some debugging, it worked. I'll show a video of the computer in action.
https://vimeo.com/800255224 

I've had a lot of fun with DLS so far, and will continue creating stuff. I have something new in the works already.

**Comment by SebLague** [comment]
That's looking awesome, thanks for sharing your work!

**Comment by Queen-Elizabeth-II** [comment]
Looking amazing there! And i have also noticed that your "map", so to say is much larger and im trying to do the same thing. any ideas on how its possible to do it again?

**Reply by zedonme** [reply]
Yes, I would like to know this too

**Reply by candymappr** [reply]
Me three

**Reply by MikZero0** [reply]
same

**Comment by Zeviraty** [comment]
Hoi andere Nederlander hoe heb je je grid vergroot want ik heb meer ruimte nodig (how do you change the grid size)
